File: ecmascript/rna-transcription/rna-transcription.js
const dna2Rna = {
  "G": "C",
  "C": "G",
  "T": "A",
  "A":"U"
}

class Transcriptor {

  toRna(dna){
    return dna.split("").map((x) => {
      if (!dna2Rna[x])
        throw new Error('Invalid input DNA.')
      return dna2Rna[x]
    }).join('')
  }
}

export default Transcriptor

File: python/rna-transcription/rna_transcription.py
def to_rna(dna):
    try:
        dna2rna = {"G": "C", "C": "G", "T": "A", "A": "U"}
        return ''.join(map(lambda x: dna2rna[x], dna))
    except KeyError:
        return ''
